sql/core/src/main/scala/org/apache/spark/sql/SparkSession.scala

```diff
@@ -213,6 +218,24 @@ class SparkSession private(
     new SparkSession(sparkContext, Some(sharedState))
```

Comment:
It's better to add `final` to avoid override this incorrectly. The method should be override is `newSession(inheritSessionState: Boolean)`.